### ISSUE TITLE
Config

### ISSUE THREAD

**Issue by tonyday567** [nebuta/d3js-haskell#1]
Hi Nebuta,

Nice start on getting d3 and haskell happening.  I needed to get some charting happening quick for a project so I went and forked, rearranging a lot of stuff.  Happy to go back and cherry pick any bits you might think worthwhile.  In terms of ideas about where to:
- the addition of a Config seemed to work well, and remving magic numbers became easy.
- is a Data2D a Scalar or a Coord or both?  It seems a bit of a tangle. Looking through the d3 tutorials it might pay to let d3 handle the translation aspects.

I have to get a few things happening with respect to functionality: line charts, mixed multi-charts, axes and ticks.  I've looked at a few solutions to doing charts in haskell and using d3 to drop straight to html seems like the best approach.

**Comment by nebuta** [comment]
Hi, thanks for picking my library. There is still so much to add in this library. My initial idea is to represent the structure of d3.js in Haskell in a type-safe way, and I have only implemented fundamental types. I guess my trial about that had some success so far :)
- Many more chart functions are still to be implemented.
- I think adding Config module is a good idea. I quickly looked at your code, and I try to incorporate it, thanks.
- Data2D is a coordinate. I don't distinguish original data from drawing coordinate in the current version, but as you say, those two should be two separate types.

I also need some tests for this library, but I'm not sure what kind of test it can have. Any idea?

**Comment by tonyday567** [comment]
When I first picked up your library, I ran through the main d3 tutorial at the same time - I had never used it before, and that seemed like a good way to `test` the code.  But I think what's needed more than tests is a gallery/tutorial.
QuickCheck will be useful for testing for sane config and parameter combinations - I'm guessing you cant have an svg that's 0 width for instance.

**Comment by nebuta** [comment]
Yes, that makes sense. So I guess you came from Haskell background more than d3.js.

**Comment by tonyday567** [comment]
I'm coming from a quant finance background, so neither.  But I'm pretty committed to haskell these days.  Let me know if there's anything I can do to help out.
